R/backport_linter.R

```diff
@@ -13,7 +13,7 @@
 backport_linter <- function(r_version = getRversion(), except = character()) {
   r_version <- normalize_r_version(r_version)
   backport_blacklist <- backports[r_version < R_system_version(names(backports))]
-  backport_blacklist <- lapply(backport_blacklist, function(nm) setdiff(nm, except))
+  backport_blacklist <- lapply(backport_blacklist, function(nm) setdiff(nm, except)) # nolint: unnecessary_lambda_linter
```

Comment:
The suggested, no lint version is less readable, IMO. Better to just skip here.

``` r
library(lintr)

r_version <- lintr:::normalize_r_version("4.0.0")
except <- character()
backports <- lintr:::backports
backport_blacklist <- backports[r_version < R_system_version(names(backports))]

lint(
  text = "lapply(backport_blacklist, function(nm) setdiff(nm, except))",
  linters = unnecessary_lambda_linter()
)
#> <text>:1:28: warning: [unnecessary_lambda_linter] Pass setdiff directly as a symbol to lapply() instead of wrapping it in an unnecessary anonymous function. For example, prefer lapply(DF, sum) to lapply(DF, function(x) sum(x)).
#> lapply(backport_blacklist, function(nm) setdiff(nm, except))
#>                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

lint(
  text = "lapply(backport_blacklist, setdiff, y = except)",
  linters = unnecessary_lambda_linter()
)

identical(
  lapply(backport_blacklist, function(nm) setdiff(nm, except)),
  lapply(backport_blacklist, setdiff, y = except)
)
#> [1] TRUE
```

<sup>Created on 2022-10-03 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
`lapply(backport_blacklist, setdiff, except)` looks fine to me. But I'm indifferent.